control-plane/agents/src/bin/core/controller/wrapper.rs

```diff
@@ -1150,13 +1152,23 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
 
     async fn shutdown_nexus(&self, request: &ShutdownNexus) -> Result<(), SvcError> {
         let dataplane = self.grpc_client_locked(request.id()).await?;
-        let result = dataplane.shutdown_nexus(request).await;
-        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
-        self.update_nexus_states(ctx.deref_mut()).await?;
-        match result {
-            // TODO: Handle idempotency
-            Ok(_) => Ok(()),
-            Err(error) => Err(error),
+        if let Some(nexus) = self.read().await.nexus(&request.uuid()) {
+            if nexus.status == NexusStatus::Shutdown {
+                Ok(())
+            } else {
+                let result = dataplane.shutdown_nexus(request).await;
+                let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+                self.update_nexus_states(ctx.deref_mut()).await?;
+                match result {
+                    // TODO: Handle idempotency.
+                    Ok(_) => Ok(()),
+                    Err(error) => Err(error),
+                }
```

Comment:
```suggestion
                result
```

Reply:
I was thinking this is where we would try to match grpc error for the request if nexus is in ShuttingDown state.